src/Datasource/EntityTrait.php

```diff
@@ -1036,12 +1036,58 @@ protected function _readError($object, $path = null)
     }
 
     /**
+     * Get a list of invalid fields and their data for errors upon validation/patching
+     *
+     * @return array
+     */
+    public function getInvalid()
```

Comment:
Maybe this also should be split into `getInvalidFields(): array` and `getInvalidField($field): mixed`?